src/client/features/enrichment/token-input.js

```diff
@@ -0,0 +1,80 @@
+const React = require('react');
+const h = require('react-hyperscript');
+const _ = require('lodash');
+const { ServerAPI } = require('../../services/');
+let Textarea = require('react-textarea-autosize').default;
+
+class TokenInput extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      query: '',
+      validTokens: [],
+      unrecognizedTokens: []
+    };
+  }
+
+  //store 'gene-input-box' contents on state
+  handleChange(e) {
+    this.state.query = e.target.value;
+  }
+
+  //call validation service API to retrieve validation result in the form of []
+  retrieveValidationAPIResult(){
+    //reset state values
+    this.state.unrecognizedTokens = [];
+    this.state.validTokens = [];
+    let tokenList = this.state.query.split(/\s/g);
+    //send all tokens to validationAPI
+    ServerAPI.enrichmentAPI({genes: _.pull(tokenList,"")}, "validation").then((result) => {
+      this.interpretValidationResult(tokenList, result.unrecognized);
+      });
+  }
+
+  //store validation data on state as arrays [validTokens] and [unrecognizedTokens]
+  //lift [validTokens] to parent file index.js
+  interpretValidationResult(tokenList, unrecognizedTokens){
+    tokenList.forEach((element) => {
```

Comment:
Is all of this necessary? 
You already have 1. unrecognized 2. recognized tokens. Also I believe you're attempting to set state on each iteration of the loop which is probably slow.